qiskit_ibm_runtime/utils/utils.py

```diff
@@ -99,6 +99,44 @@ def validate_isa_circuits(circuits: Sequence[QuantumCircuit], target: Target) ->
             )
 
 
+def is_static_circuit(circuit: QuantumCircuit) -> bool:
```

Comment:
Great, I'll switch.

Reply:
I see that `are_circuits_dynamic` also handles the case where the circuit is a string by simply returning `True` (although the type deceleration is for `QuantumCircuit` only). This might cause a problem in the future, if `qiskit-ibm-runtime` supports string circuits in pubs (already supported server-side).

I suggest something like attempting conversion of the string to circuit using the standard `qiskit` methods before performing the check, or performing keyword search inside the given string (less robust, more efficient).

Reply:
Didn't see your comment before I made my other one. But we can just remove the `str` as an input for now, since we don't (yet) officially support OpenQASM3 as an input type for circuits in qiskit-ibm-runtime

Reply:
Currently we also support `backend.run` which accepts OpenQASM3 and removing the `str` inputs breaks this (and the test `test_single_openqasm3_submission`). Given that `backend.run` is about to be deprecated I suggest returning `are_circuits_dynamic` to its previous state + opening an issue for its future change.

Reply:
Good point. We can have `are_circuits_dynamic` return False if the input is QASM for now. If we return True then people can't use DD even if the circuit is not dynamic.